Modules/IO/TIFF/src/itkTIFFImageIO.cxx

```diff
@@ -456,21 +456,33 @@ TIFFImageIO::ReadImageInformation()
       this->SetPixelType(RGBA);
   }
 
+  bool isPaletteShortType = false;
   if ((this->GetFormat() == TIFFImageIO::PALETTE_GRAYSCALE || this->GetFormat() == TIFFImageIO::PALETTE_RGB) &&
       m_TotalColors > 0 && this->GetExpandRGBPalette())
+
   {
-    m_ComponentType = UCHAR;
     // detect if palette appears to be 8-bit or 16-bit
     for (unsigned int cc = 0; cc < 256; ++cc)
```

Comment:
Although the 256 the 8-bit grayscale values can be identified automatically, I am wondering whether we could use `UCHAR_MAX` instead, or 2^sizeof(UCHAR). @hjmjohnson, @thewtex Is the proposal adding more generalization than necessary? Should may be done in another PR?

Reply:
I'm not sure to understand the problem, I hope I'm not missunderstanding you, but here, the first 256 correspond to the maximum number of colors in the palette of a tiff file. Tiff palette is saved in the file as 256 x 3 integers of  2^16 bits whether they are actually unsigned char or not (https://www.awaresystems.be/imaging/tiff/tifftags/colormap.html) so I guess it doesn't depends on the platform.

Reply:
@YannLePoul thanks for the explanation. Looks like I was then confused by the `m_ComponentType = UCHAR;` statement that your PR removes. Reading the comment line just above clarifies things :sweat_smile: 

Resolving the conversation.